### PR TITLE
feat: [M3-7577] - Add invoice byline for powered down instances

### PR DESCRIPTION
**Pull request by jaalah-akamai**
## Description 📝
In an effort to clarify to customers that they are being charged for powered-down (shutdown) instances, we want to add the following byline to customer invoices:
```
This invoice may include Linode Compute Instances that have been powered off as the data is maintained and resources are still reserved. If you no longer need powered-down Linodes, you can <a href="https://www.linode.com/docs/products/platform/billing/guides/stop-billing/">remove the service</a> from your account.
```
>[!note]
Adding a link in the PDF added a lot of complexity, so I reached out to Customer Success and they seemed fine with spelling out the link. Open to suggestions to make that better.

## Changes  🔄
- Updated generated PDF with logic to smartly break to new page if content doesn't fit
- Added copy to Cloud Manager invoice page

## Target release date 🗓️
3/4/2024

## Preview 📷
| Desc      | Screenshot |
| ----------- | ----------- |
| Cloud Manager |Screenshot 2024-02-20 at 9 46 33 AM|
|PDF|Screenshot 2024-02-20 at 12 15 24 PM|

### Verification steps
(How to verify changes)
- Go to `/account/billing` and look at any invoice
- Download the PDF and ensure text is there too.
- You can turn on MSW and update the buildList for `*invoices/:invoiceId/items` to check various heights

```
rest.get('*invoices/:invoiceId/items', (req, res, ctx) => {
    const items = invoiceItemFactory.buildList(10);
    return res(ctx.json(makeResourcePage(items, { page: 1, pages: 1 })));
  }),
```

## As an Author I have considered 🤔

*Check all that apply*

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/features/Billing/PdfGenerator/utils.ts [hunk 1]
```diff
@@ -210,6 +210,8 @@ const getTaxSummaryBody = (taxSummary: TaxSummary[]) => {
  * Creates the totals table for Invoice PDF
  */
 export const createInvoiceTotalsTable = (doc: JSPDF, invoice: Invoice) => {
+  let finalY = 0; // Initialize a variable to hold the final Y position after the table is drawn
```
Comment (jdamore-linode) [inline packages/manager/src/features/Billing/PdfGenerator/utils.ts:213]: Can this be declared in the `didDrawPage` hook?

**Review by jdamore-linode — APPROVED** [review]
This is awesome! Confirmed that the blurb appears at the bottom of the last page of the invoice -- wasn't able to find an invoice with the right amount of data to properly test the new page logic, but the code looks good to me

**Comment by jaalah-akamai** [comment]
This is good for final approvals/review, I added `wait to merge` for some additional internal review as well.

**Review by bnussman-akamai — APPROVED** [review]
Screenshot 2024-02-21 at 9 17 54 AM

File: packages/manager/src/features/Billing/PdfGenerator/utils.ts [hunk 2]
```diff
@@ -233,6 +233,30 @@ export const createInvoiceTotalsTable = (doc: JSPDF, invoice: Invoice) => {
...
     },
+    didDrawPage: (data) => {
+      let finalY = 0; // Initialize a variable to hold the final Y position after the table is drawn
+
+      if (data?.cursor?.y) {
+        finalY = data.cursor.y;
+      }
+
+      const footerText =
+        'This invoice may include Linode Compute Instances that have been powered off as the data is maintained and resources are still reserved. If you no longer need powered-down Linodes, you can remove the service (https://www.linode.com/docs/products/platform/billing/guides/stop-billing/) from your account.';
```
Comment (carrillo-erik) [inline packages/manager/src/features/Billing/PdfGenerator/utils.ts:244]: Regarding your note about adding a link in the PDF, have we considered switching PDF libraries? The last version of `jspdf` was published 2 years ago, whereas, something like [react-pdf/renderer](https://www.npmjs.com/package/@react-pdf/renderer) was published 15 days ago and the package size is smaller by about 1/7.

**Review by carrillo-erik — APPROVED** [review]

**Pull request merged**